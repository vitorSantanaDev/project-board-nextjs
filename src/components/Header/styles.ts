import styled from 'styled-components'

export const Wrapper = styled.header`
  height: 8rem;
  background-color: #141a29;
`

export const HeaderContent = styled.div`
  height: 8rem;
  max-width: 1120px;

  margin: 0 auto;

  display: flex;
  align-items: center;
`

export const Logo = styled.img`
  cursor: pointer;
`

export const Navigation = styled.nav`
  height: 5rem;
  margin-left: 5rem;

  a {
    position: relative;
    display: inline-block;

    padding: 0 0.5rem;

    height: 5rem;
    line-height: 5rem;

    color: #fff;
    font-size: 1.8rem;

    & + a {
      margin-left: 2rem;
    }
  }
`
