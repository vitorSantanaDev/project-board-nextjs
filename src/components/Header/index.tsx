import SigInButton from '../SigInButton'
import Link from 'next/link'
import * as S from './styles'

const Header: React.FC = () => {
  return (
    <S.Wrapper>
      <S.HeaderContent>
        <Link href="/">
          <S.Logo
            src="/images/small-logo.svg"
            alt="Logo meu Board, letra B com circulo amarelo ao lado"
          />
        </Link>
        <S.Navigation>
          <Link href="/">Home</Link>
          <Link href="/board">Meu Board</Link>
        </S.Navigation>

        <SigInButton session={true} />
      </S.HeaderContent>
    </S.Wrapper>
  )
}

export default Header
