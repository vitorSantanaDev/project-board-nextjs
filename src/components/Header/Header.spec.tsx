import { render, screen } from '@testing-library/react'
import Header from '.'

describe('<Header />', () => {
  it('should render the background color correctly', () => {
    const { container } = render(<Header />)
    expect(container.firstChild).toHaveStyle({ 'background-color': '#141A29' })

    screen.logTestingPlaygroundURL()
  })

  it('should render logo Borad', () => {
    render(<Header />)
    const logo = screen.getByRole('img', {
      name: /logo meu Board, letra B com circulo amarelo ao lado/i
    })
    expect(logo).toBeInTheDocument()
  })

  it('should render links', () => {
    render(<Header />)

    const navLinkHome = screen.getByRole('link', { name: /home/i })
    expect(navLinkHome).toBeInTheDocument()
    expect(navLinkHome).toHaveAttribute('href', '/')
    expect(navLinkHome).toHaveStyle({ color: '#FFF' })

    const navLinkMyBoard = screen.getByRole('link', { name: /meu board/i })
    expect(navLinkMyBoard).toBeInTheDocument()
    expect(navLinkMyBoard).toHaveAttribute('href', '/board')
    expect(navLinkMyBoard).toHaveStyle({ color: '#FFF' })
  })

  it('should render the component Button', () => {
    render(<Header />)
    const sigInButton = screen.getByRole('button')
    expect(sigInButton).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<Header />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
