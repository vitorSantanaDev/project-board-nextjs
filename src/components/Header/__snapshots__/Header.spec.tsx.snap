// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c4 {
  height: 2.5rem;
  border-radius: 3rem;
  cursor: pointer;
  padding: 0 1.5rem;
  margin-left: auto;
  color: #fff;
  background-color: transparent;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: ease-in-out 0.3s;
  transition: ease-in-out 0.3s;
}

.c4 img {
  width: 45px;
  height: 45px;
  margin-right: 1.2rem;
}

.c4 svg {
  width: 30px;
  height: 30px;
}

.c4 svg:first-child {
  margin-right: 1rem;
}

.c4 svg.logoutIcon {
  margin-left: 1.2rem;
}

.c4:hover {
  opacity: 0.8;
}

.c0 {
  height: 8rem;
  background-color: #141a29;
}

.c1 {
  height: 8rem;
  max-width: 1120px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  cursor: pointer;
}

.c3 {
  height: 5rem;
  margin-left: 5rem;
}

.c3 a {
  position: relative;
  display: inline-block;
  padding: 0 0.5rem;
  height: 5rem;
  line-height: 5rem;
  color: #fff;
  font-size: 1.8rem;
}

.c3 a + a {
  margin-left: 2rem;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Logo meu Board, letra B com circulo amarelo ao lado"
      class="c2"
      src="/images/small-logo.svg"
    />
    <nav
      class="c3"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/board"
      >
        Meu Board
      </a>
    </nav>
    <button
      class="c4"
      type="button"
    >
      <img
        alt="Photo do usário logado"
        src="/images/person-exemple.png"
      />
      Olá Harry
      <svg
        class="logoutIcon"
        color="#737380"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(115, 115, 128);"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
        />
        <polyline
          points="16 17 21 12 16 7"
        />
        <line
          x1="21"
          x2="9"
          y1="12"
          y2="12"
        />
      </svg>
    </button>
  </div>
</header>
`;
